flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClient.java

```diff
@@ -171,10 +171,7 @@ public void operationComplete(ChannelFuture future) throws Exception {
 	}
 
```

Comment:
Hmmm all of the changes in `[FLINK-XXXXX][network] reduce locking around credit notification` seems shaky. Removing the code that's commented "do sth to avoid deadlock" seems adventurous :)